specs/PeriodicBackgoundSyncAndBackgroundSync.md

```diff
@@ -0,0 +1,581 @@
+                                Microsoft::WRL::Callback<
+                                    ICoreWebView2SyncRegisteredEventHandler>(
+                                    [this](
+                                        ICoreWebView2ServiceWorkerSyncRegistrationManager*
+                                            sender,
+                                        ICoreWebView2ServiceWorkerSyncRegisteredEventArgs*
+                                            args)
+                                    {
+                                        wil::com_ptr<
+                                            ICoreWebView2ServiceWorkerSyncRegistrationInfo>
+                                            syncRegistrationInfo;
+                                        CHECK_FAILURE(
+                                            args->get_RegistrationInfo(&syncRegistrationInfo));
+                                        std::wstringstream message;
+                                        AppendSyncRegistrationInfo(
+                                            syncRegistrationInfo, true, message);
+                                        m_appWindow->AsyncMessageBox(
+                                            message.str(), L"Periodic Background Sync");
+                                        return S_OK;
+                                    })
+                                    .Get(),
+                                &m_periodicSyncRegisteredToken));
+                            //! [PeriodicSyncRegistered]
+                        }
+                        //! [SyncRegistrationManager]
+                    }
+                    return S_OK;
+                })
+                .Get()));
+    }
+
+    // Receive dispatch periodic sync task request
+    CHECK_FAILURE(m_webView->add_WebMessageReceived(
+        Callback<ICoreWebView2WebMessageReceivedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2WebMessageReceivedEventArgs* args)
+                -> HRESULT
+            {
+                wil::unique_cotaskmem_string source;
+                CHECK_FAILURE(args->get_Source(&source));
+                wil::unique_cotaskmem_string webMessageAsString;
+                if (SUCCEEDED(args->TryGetWebMessageAsString(&webMessageAsString)))
+                {
+                    if (wcscmp(source.get(), m_sampleUri.c_str()) == 0)
+                    {
+                        std::wstring message = webMessageAsString.get();
+                        if (message.compare(0, 26, L"DispatchPeriodicSyncEvent ") == 0)
+                        {
+                            DispatchPeriodicBackgroundSyncTask(message.substr(26).c_str());
+                        }
+                        else if (wcscmp(message.c_str(), L"GetPeriodicSyncRegistrations") == 0)
+                        {
+                            ShowAllSyncRegistrationInfos(COREWEBVIEW2_SERVICE_WORKER_SYNCHRONIZATION_KIND_PERIODIC_SYNC);

```

Comment:
Same question as previous about the scenario for the sample code. This looks like sandbox / playground sort of code to play with the API rather than sample code that demonstrates something an end dev would actually do with this API in their code.